documentation/project-docs/breaking-change-guidelines.md

```diff
@@ -0,0 +1,81 @@
+    * For example, `--configuration` flag defaulting to `Release` instead of `Debug`
+
+## Configuration Knobs
+
+The following knobs are available to enable/disable these changes (some may not apply to all kinds of changes):
+
+* [TargetFramework](https://learn.microsoft.com/en-us/dotnet/standard/frameworks)
+* [SdkAnalysisLevel](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#sdkanalysislevel)
+* EditorConfig (for Analyzers)
+* AnalysisLevel (for Analyzers/CodeFixes)
+* WarningLevel
+
+## Deciding how to Surface Changes
+
+### Implement changes in an informational/non-blocking way initially
+
+What this means will vary change-to-change. For example, for a change expressed as an Analyzer or MSBuild diagnostic, consider
+Informational level severities initially. For a behavioral change on a CLI, consider an informational message written to the
+stderr channel on the console instead of making the stdout output un-parseable by tools.
+
+Concrete example: When the `dotnet new --list` command was changed to `dotnet new list` to adhere to CLI design guidelines,
+the old command was still the default supported and a warning was written to the console when the old form was used pointing users to the new form. This allowed users and scripts to continue using the old form and gradually migrate to the new.
+
+### Gradually increase severity over each release
+
+If a change in introduced in an informational/non-blocking way, determine the time frame where it is safe to increase the severity. For Analyzers, this may mean tying it to the next value of AnalysisLevel (which is downstream of TFM). For small MSBuild and NuGet diagnostics, this may mean tying it to the next Warning Level or SdkAnalysisLevel. For CLI changes, this may mean tying it to the next LTS major version of the SDK. Ideally the way you would structure this increase would be automated and documented so that users know what's coming down the pipe.
+
+### Cut-over to new behavior after a long introduction period
+
+After the change has been introduced in a gradual way, cut over to the new behavior. This may mean removing the old behavior entirely, or it may mean making the new behavior the default and providing a way to opt out of it. It is important that you
+provide enough time for users to adapt - for example the `dotnet new --list` example above took an entire major release to make the new forms the default.
+
+### Always provide a way to opt out of the change
+
+Have some kind of knob that allows users to opt out of the change entirely. This could be a flag, an environment variable, or a global.json setting. This allows users to continue using the old behavior if they need to in exceptional situations. It is
+important to document this knob and its behavior in the SDK documentation. It is also important to define a timeline for when this knob will be removed entirely, forcing users to adopt the new behavior.
```

Comment:
fwiw, I might never remove some knobs if they are simple properties or env variables. I don't think Roslyn removes langver support but we should check with them.

Reply:
ahh, you mention that may be never for analyzers below. I might include items that have a simple on/off.

Reply:
I removed the break line.